lib/src/select/types.ts

```diff
@@ -103,43 +103,14 @@ type CommonProps = {
    */
   tabIndex?: number;
 };
-type SingleSelect = CommonProps & {
-  /**
-   * If true, the select component will support multiple selected options.
-   * In that case, value will be an array of strings with each selected
-   * option value.
-   */
-  multiple?: false;
-  /**
-   * Initial value of the select, only when it is uncontrolled.
-   */
-  defaultValue?: string;
-  /**
-   * Value of the select. If undefined, the component will be uncontrolled
-   * and the value will be managed internally by the component.
-   */
-  value?: string;
-  /**
-   * This function will be called when the user selects an option.
-   * An object including the current value and the error (if the value entered is not valid)
-   * will be passed to this function. If there is no error, error will not be defined.
-   */
-  onChange?: (val: { value: string; error?: string }) => void;
-  /**
-   * This function will be called when the select loses the focus. An
-   * object including the value and the error (if the value
-   * selected is not valid) will be passed to this function. If there is no error,
-   * error will not be defined.
-   */
-  onBlur?: (val: { value: string; error?: string }) => void;
-};
-type MultipleSelect = CommonProps & {
+
+type Props = CommonProps & {
   /**
    * If true, the select component will support multiple selected options.
    * In that case, value will be an array of strings with each selected
    * option value.
    */
-  multiple: true;
+  multiple?: boolean;
```

Comment:
We should stick to the previous implementation with `SingleSelect` and `MultipleSelect` types. The point of how it was done was to guide the user through the construction of a Select using the types. For example, if `multiple` is true, the `defaultValue` should be an array of strings.

The solution should only be in the Select.tsx files, not in the current types.

Reply:
The problem is that, using those types, for some reason both the onChange and the onBlur callback seem to be inferred as the following: 

```
(val: {
    value: string;
    error?: string;
} & {
    value: string[];
    error?: string;
}) => void
```
Instead of the expected:
```
((val: {
    value: string;
    error?: string;
}) => void) | ((val: {
    value: string[];
    error?: string;
}) => void)
```

I have tried the approach recommended here: [Conditional Types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) among others recommended in forums with no success for this case.

Reply:
We have decided that the best and easiest approach would be to use type assertion. [More info here](https://stackoverflow.com/questions/76188330/typescript-union-type-of-functions-weirdly-becomes-an-intersection-type-of-the-a)